hyper/http20/connection.py

```diff
@@ -740,6 +742,12 @@ def _recv_cb(self, stream_id=0):
                 self.recent_recv_streams.discard(stream_id)
                 return
 
+            # make sure to validate the stream is readable.
+            # if the connection was reset, this stream id won't appear in
+            # self.streams and will cause this call to raise an exception.
+            if stream_id:
```

Comment:
one way I could make it more clear is something like:

```
if stream_id != 0:
```

What kind of values do we see as stream ids? Are they sequential integers always? This could be a problem with my other check where I bail if the stream_id isnt in self.streams because the new socket connection could generate another stream id that is another integer that is identical to the one we currently have. I need to read the http2 spec.